src/main/kotlin/org/neo4j/graphql/SchemaBuilder.kt

```diff
@@ -0,0 +1,299 @@
+import org.neo4j.graphql.handler.*
+import org.neo4j.graphql.handler.projection.ProjectionBase
+import org.neo4j.graphql.handler.relation.CreateRelationHandler
+import org.neo4j.graphql.handler.relation.CreateRelationTypeHandler
+import org.neo4j.graphql.handler.relation.DeleteRelationHandler
+
+object SchemaBuilder {
+
+    @JvmStatic
+    @JvmOverloads
+    fun buildSchema(sdl: String, config: SchemaConfig = SchemaConfig()): GraphQLSchema {
+        val schemaParser = SchemaParser()
+        val typeDefinitionRegistry = schemaParser.parse(sdl)
+
+        val builder = RuntimeWiring.newRuntimeWiring()
+
+        AugmentationProcessor(typeDefinitionRegistry, config, builder).augmentSchema()
+
+        val runtimeWiring = builder.build()
+
+        // todo add new queries, filters, enums etc.
+        return SchemaGenerator().makeExecutableSchema(typeDefinitionRegistry, runtimeWiring)
+    }
+
+    private class AugmentationProcessor(
+            val typeDefinitionRegistry: TypeDefinitionRegistry,
+            val schemaConfig: SchemaConfig,
+            val wiringBuilder: RuntimeWiring.Builder
+    ) {
+        private val metaProvider = TypeRegistryMetaProvider(typeDefinitionRegistry)
+        private var schemaDefinition: SchemaDefinition
+        private var queryDefinition: ObjectTypeDefinition
+        private var mutationDefinition: ObjectTypeDefinition
+
+        init {
+            this.schemaDefinition = typeDefinitionRegistry
+                .schemaDefinition()
+                .orElseGet { SchemaDefinition.newSchemaDefinition().build() }
+            this.queryDefinition = getOrCreateObjectTypeDefinition("Query")
+            this.mutationDefinition = getOrCreateObjectTypeDefinition("Mutation")
+        }
+
+        fun augmentSchema() {
+            mergeNeo4jEnhancements()
+
+            val interfaceTypeDefinitions = typeDefinitionRegistry.types().values
+                .filterIsInstance<InterfaceTypeDefinition>()
+                .map { enhanceRelations(it, InterfaceTypeDefinition.CHILD_DEFINITIONS) }
+            val objectTypeDefinitions = typeDefinitionRegistry.types().values
+                .filterIsInstance<ObjectTypeDefinition>()
+                .filter { !it.isNeo4jType() }
+                .map { enhanceRelations(it, ObjectTypeDefinition.CHILD_FIELD_DEFINITIONS) }
+
+            wireDefinedOperations(queryDefinition, true)
+            wireDefinedOperations(mutationDefinition, false)
+
+            val relationTypes = objectTypeDefinitions
+                .filter { it.getDirective(DirectiveConstants.RELATION) != null }
+                .map { it.getDirective(DirectiveConstants.RELATION).getArgument(DirectiveConstants.RELATION_NAME).value.toJavaValue().toString() to it }
+                .toMap()
+
+            val nodeDefinitions: List<TypeDefinition<*>> = interfaceTypeDefinitions + objectTypeDefinitions
+            nodeDefinitions.forEach { createNodeMutation(it.getNodeType()!!) }
```

Comment:
what does it mean to create an interface?

Reply:
You can update / merge / delete interfaces , but not create them